### PR TITLE
chore: Deprecation for ExpressCheckoutService in 4.0

### LINKED ISSUE #12855: Express checkout deprecated constructore
Express checkout have deprecated constructor in new checkout lib

### PR DESCRIPTION
**Pull request by dydome**
close #12855

### CODE REVIEW

File: projects/schematics/src/shared/constants.ts [hunk 1]
```diff
@@ -41,6 +41,9 @@ export const SPARTACUS_STOREFINDER = '@spartacus/storefinder';
 export const SPARTACUS_TRACKING = '@spartacus/tracking';
 export const SPARTACUS_USER = '@spartacus/user';
 export const SPARTACUS_CHECKOUT = '@spartacus/checkout';
+export const SPARTACUS_CHECKOUT_CORE = `${SPARTACUS_CHECKOUT}/administration/core`;
+export const SPARTACUS_CHECKOUT_ROOT = `${SPARTACUS_CHECKOUT}/administration/root`;
```
Comment (marlass) [inline projects/schematics/src/shared/constants.ts]: administration? 👀

File: feature-libs/checkout/components/services/express-checkout.service.ts [hunk 2]
```diff
@@ -286,22 +286,9 @@ export class ExpressCheckoutService {
     );
   }
 
-  /**
-   * @deprecated since version 3.2
-   * Use ClearCheckoutService to clear the checkout state
-   */
-  protected resetCheckoutProcesses() {
```
Comment (marlass) [inline feature-libs/checkout/components/services/express-checkout.service.ts:293]: Also add method-property migration for this method and mention it in the docs

**Review by marlass — CHANGES_REQUESTED** [review]

File: projects/schematics/src/migrations/4_0/methods-and-properties-deprecations/data/express-checkout.service.migration.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+} from '../../../../shared/constants';
+import { MethodPropertyDeprecation } from '../../../../shared/utils/file-utils';
+
+export const EXPRESS_CHECKOUT_SERVICE_MIGRATION: MethodPropertyDeprecation[] = [
+  // feature-libs/checkout/components/services/express-checkout.service.ts
+  {
+    class: EXPRESS_CHECKOUT_SERVICE,
+    importPath: SPARTACUS_CHECKOUT_CORE,
+    deprecatedNode: RESET_CHECKOUT_PROCESSES,
+    comment: `// ${TODO_SPARTACUS} Method '${RESET_CHECKOUT_PROCESSES}' was removed, use '${CLEAR_CHECKOUT_SERVICE}' method instead`,
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/methods-and-properties-deprecations/data/express-checkout.service.migration.ts]: Add it also to the migration doc

**Review by marlass — CHANGES_REQUESTED** [review]

File: docs/migration/4_0.md [hunk 4]
```diff
@@ -836,6 +836,12 @@ The following properties where removed in 4.0:
 
 - `CurrentUnitService` is now required parameter in component constructor.
 
+
+#### ExpressCheckoutService
+
+- `ClearCheckoutService` is now required parameter in component constructor.
+- Method `resetCheckoutpProcesses` was removed, use `clearCheckoutService` method instead.
```
Comment (marlass) [inline docs/migration/4_0.md]:
```suggestion
- Method `resetCheckoutProcesses` was removed, use method `resetCheckoutProcesses` from `ClearCheckoutService` instead.
```

File: projects/schematics/src/migrations/4_0/methods-and-properties-deprecations/data/express-checkout.service.migration.ts (hunk 3 above)
Comment (marlass) [inline projects/schematics/src/migrations/4_0/methods-and-properties-deprecations/data/express-checkout.service.migration.ts]:
```suggestion
    comment: `// ${TODO_SPARTACUS} Method '${RESET_CHECKOUT_PROCESSES}' was removed, use method '${RESET_CHECKOUT_PROCESSES}' from '${CLEAR_CHECKOUT_SERVICE}' instead`,
```

**Review by marlass — CHANGES_REQUESTED** [review]

**Review by marlass — DISMISSED** [review]

**Review by marlass — DISMISSED** [review]

**Comment by dydome** [comment]
Currently suspended. Architects' analysis needed, waiting for @marlasak  decision.

File: projects/schematics/src/migrations/4_0/constructor-deprecations/data/express-checkout.service.migration.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  SPARTACUS_CORE,
+  USER_ADDRESS_SERVICE,
+  USER_PAYMENT_SERVICE,
+} from '../../../../shared/constants';
+import { ConstructorDeprecation } from '../../../../shared/utils/file-utils';
+
+export const EXPRESS_CHECKOUT_SERVICE_MIGRATION: ConstructorDeprecation = {
+  // feature-libs/checkout/components/services/express-checkout.service.ts
+  class: EXPRESS_CHECKOUT_SERVICE,
+  importPath: SPARTACUS_CHECKOUT_CORE,
```
Comment (znikola) [inline projects/schematics/src/migrations/4_0/constructor-deprecations/data/express-checkout.service.migration.ts]: According to the written 4_0.md file, the `express-checkout.service.ts` is now in the `@spartacus/checkout/components`, while the `SPARTACUS_CHECKOUT_CORE` constants points to `@spartacus/checkout/core`

File: projects/schematics/src/migrations/4_0/rename-symbol/rename-symbol.ts [hunk 6]
```diff
@@ -1,16 +1,29 @@
 import { Rule, Tree } from '@angular-devkit/schematics';
+import { SPARTACUS_CHECKOUT_ROOT } from 'feature-libs/checkout/schematics/constants';
```
Comment (znikola) [inline projects/schematics/src/migrations/4_0/rename-symbol/rename-symbol.ts]: Please fix the import

**Review by znikola — CHANGES_REQUESTED** [review]

**Review by marlass — APPROVED** [review]

**Pull request merged**